support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -744,7 +760,7 @@ function CheckoutComponent({ geoId }: Props) {
 					paymentMethod: paymentMethod,
 				};
 				setThankYouOrder(order);
-				window.location.href = `/${geoId}/thank-you`;
+				window.location.href = `/${geoId}/thank-you?product=${productId}&ratePlan=${query.ratePlan}&promoCode=${promoCode}`;
```

Comment:
This data [is being set](https://github.com/guardian/support-frontend/blob/main/support-frontend/assets/pages/%5BcountryGroupId%5D/checkout.tsx#L746) in the [`localStorage` `order` object](https://github.com/guardian/support-frontend/blob/main/support-frontend/assets/pages/%5BcountryGroupId%5D/thank-you.tsx#L56-L77).

The reasoning is that the thank you page is based on the state of a complete order, which is reset when trying to make a new order.

Do you think this is easier to reason about? If so, it would be good to remove the `localStorage` implementation all together. 

If we want to keep that solution maybe this is a sign we should signpost it better.

Reply:
Personally I prefer query strings in general because then the url contains everything it needs to render the page. Otherwise the page can be broken if the correct state is not set in local storage but it won't be obvious why, it's like the difference between a pure function and one with side effects. Having said that, if you think the local storage solution works better in this case I'm happy to change it.

Reply:
Indeed - I just thought the page was specific to your order, hence the differentiation. I like the idea of keeping the state URL driven - and the worst I can see happening is someone changing the URL params and seeing the page change...